### PR TITLE
1985 clock switch

### PR DESCRIPTION
**Pull request by muryno**
#1985 

@kylecorry31 , I hide the title digital time when in digital mode, instead of showing twice on the screen

https://github.com/kylecorry31/Trail-Sense/assets/26343440/c6f4e4e2-5294-436a-bd74-0fb904deb4d4

### CODE REVIEW

File: app/src/main/java/com/kylecorry/trail_sense/tools/clock/ui/ClockView.kt [hunk 1]
```diff
@@ -59,12 +60,36 @@ class ClockView : View {
...
+            drawHands(canvas)
+            drawCenter(canvas)
+        } else {
+            drawDigitalClock(canvas)
+        }
         postInvalidateDelayed(20)
         invalidate()
     }
+    private fun drawDigitalClock(canvas: Canvas) {
+        val timeText = if (use24Hours) {
```
Comment (kylecorry31) [inline app/src/main/java/com/kylecorry/trail_sense/tools/clock/ui/ClockView.kt:74]: Can this use FormatService.formatTime?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1404,4 +1406,6 @@
     <!--    Maximum value (ex. Max 100 ft)-->
     <string name="max_s">Max %s</string>
     <string name="ar_calibration_disclaimer">Before each use of the Augmented Reality tool, verify the sun or moon icons are properly aligned to the real world. If not, click the compass icon in the top left of the screen and follow the instructions to calibrate.</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (kylecorry31) [inline app/src/main/res/values/strings.xml:1409]: This should be removed

File: app/src/main/java/com/kylecorry/trail_sense/tools/clock/ui/ClockView.kt (hunk 1 above)
Comment (muryno) [reply app/src/main/java/com/kylecorry/trail_sense/tools/clock/ui/ClockView.kt:76]: @kylecorry31 updated

**Comment by kylecorry31** [comment]
I just ran this on my phone and the text is cutting off (both sides). 

Also, can this just be a TextView rather than drawing onto the canvas? That way that even on a small screen phone it will wrap to a new line if needed.

**Comment by muryno** [comment]
> I just ran this on my phone and the text is cutting off (both sides).
> 
> Also, can this just be a TextView rather than drawing onto the canvas? That way that even on a small screen phone it will wrap to a new line if needed.
> 
> Screenshot_20240414-081433

HHmmm....  So we need to have another textView in  and swap when it is analog instead of moving the digital logic to ClockView?

> I just ran this on my phone and the text is cutting off (both sides).
> 
> Also, can this just be a TextView rather than drawing onto the canvas? That way that even on a small screen phone it will wrap to a new line if needed.
> 
> Screenshot_20240414-081433

HHmm so in that case we need to have a textView in ToolClockFragment.xml that will be hidden 

> I just ran this on my phone and the text is cutting off (both sides).
> 
> Also, can this just be a TextView rather than drawing onto the canvas? That way that even on a small screen phone it will wrap to a new line if needed.
> 
> Screenshot_20240414-081433

Instead of creating a textView, I am thinking of getting the width of the ClockView container and make the text has padding horizontally... should be able to react to screen size dynamically

**Comment by muryno** [comment]
I am thinking of updating ClockView by getting the width of the container and making the text has horizontal padding...

Your suggested approach will take the logic for  digital clockout of ClockView. 

what do you think?

**Comment by muryno** [comment]
now updated 



Screenshot_20240414_152347_Trail Sense

**Comment by kylecorry31** [comment]
> I am thinking of updating ClockView by getting the width of the container and making the text has horizontal padding...
> 
> Your suggested approach will take the logic for digital clock out of ClockView.

Yes, I believe moving it out of ClockView will be better (so ClockView will only be for the analog clock). The SP units are adjustable the the user on the device via text scaling, so it's possible this will still clip (and with the default text scaling, I can get it to clip on my emulator). The textview also supports autosizing the text to fit within the bounds (with a minimum and maximum size: https://developer.android.com/develop/ui/views/text-and-emoji/autosizing-textview), which I think should be used.

Here's what your current change looks like on my emulator:
image

**Comment by muryno** [comment]
Ok can make the ClockView only handles analog view, then have a textview that hide/show base on users preference

**Comment by muryno** [comment]
@kylecorry31  Updated now

**Comment by kylecorry31** [comment]
Looks good, thanks!

**Pull request merged**